web/docs/guides/middleware-customization.md

```diff
@@ -0,0 +1,134 @@
+
+export type MiddlewareConfigFn = (middleware: MiddlewareConfig) => MiddlewareConfig
+
+const defaultGlobalMiddleware: MiddlewareConfig = new Map([
+  ['helmet', helmet()],
+  ['cors', cors({ origin: config.allowedCORSOrigins })],
+  ['logger', logger('dev')],
+  ['express.json', express.json()],
+  ['express.urlencoded', express.urlencoded({ extended: false })],
+  ['cookieParser', cookieParser()]
+])
+```
+
+Here is the [definition of config](https://github.com/wasp-lang/wasp/blob/main/waspc/data/Generator/templates/server/src/config.js).
+
+## 1. Customize global middleware
+
+If you would like to modify the middleware for _all_ operations and APIs, you can do something like:
+
+```c title=todoApp.wasp
+app todoApp {
+  // ...
+
+  server: {
+    setupFn: import setup from "@server/serverSetup.js",
+    middlewareConfigFn: import { serverMiddlewareFn } from "@server/serverSetup.js"
+  },
+}
+```
+
+```ts title=src/server/serverSetup.js
+import cors from 'cors'
+import { MiddlewareConfigFn } from '@wasp/middleware'
+import config from '@wasp/config.js'
+
+export const serverMiddlewareFn: MiddlewareConfigFn = (middleware) => {
+  // Example of adding an extra domains to CORS.
+  middleware.set('cors', cors({ origin: [config.frontendUrl, 'https://example1.com', 'https://example2.com'] }))
+  return middleware
+}
+```
+
+## 2. Customize `api`-specific middleware
+
+If you would like to modify the middleware for a single API, you can do something like:
+
+```c title=todoApp.wasp
+api webhookCallback {
+  fn: import { webhookCallback } from "@server/apis.js",
+  middlewareConfigFn: import { webhookCallbackMiddlewareFn } from "@server/apis.js",
+  httpRoute: (POST, "/webhook/callback"),
+  auth: false
+}
+```
+
+```ts title=src/server/apis.ts
+import express from 'express'
+import { WebhookCallback } from '@wasp/apis/types'
+import { MiddlewareConfigFn } from '@wasp/middleware'
+
+export const webhookCallback: WebhookCallback = (req, res, _context) => {
+  res.json({ msg: req.body.length })
+}
+
+export const webhookCallbackMiddlewareFn: MiddlewareConfigFn = (middleware) => {
+  console.log('webhookCallbackMiddlewareFn: Swap express.json for express.raw')
+  
+  middleware.delete('express.json')
+  middleware.set('express.raw', express.raw({ type: '*/*' }))
+
+  return middleware
+}
+
+```
+
+:::note
+This gets installed on a per-method basis. Behind the scenes, this results in something like:
+
+```js
+router.post('/webhook/callback', webhookCallbackMiddleware, ...)
+```
+:::
+
+## 3. Customize path-path middleware
+
+If you would like to modify the middleware for all API routes under some common path, you can do something like:
+
+```c title=todoApp.wasp
+namespace fooBar {
+  middlewareConfigFn: import { fooBarNamespaceMiddlewareFn } from "@server/apis.js",
+  path: "/foo/bar"
+}
+```
+
+:::note
+The JS/TS implementation is not shown, as it would be similar to the examples above.
```

Comment:
Why not? For completeness sake and if somebody links to this piece directly, they didn't really read the previous part.